### PR TITLE
CDM (Synapse Link for Dynamics) plugin

### PR DESCRIPTION
**Pull request by george-zubrienko**
CDM Plugin WIP

### CODE REVIEW

File: framework/arcane-framework/src/main/scala/services/cdm/CdmTable.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+   * Read a table snapshot, taking optional start time.
+   * @param fromYears Folders from Synapse export to include in the snapshot. If not provided, ALL folders will be included
+   * @return A stream of rows for this table
+   */
+  def snapshot(fromYears: Option[Int]): Future[LazyList[DataRow]] =
+    // list all matching blobs
+    Future.sequence(getListPrefixes(fromYears)
+      .flatMap(prefix => reader.listBlobs(storagePath + prefix))
+      .map(blob => reader.getBlobContent(storagePath + blob.name, _.map(_.toChar).mkString)))
+      .map(_.flatMap(content => content.split('\n').foldLeft((Seq.empty[String], "")) { (agg, value) =>
```
Comment (s-vitaliy) [inline framework/arcane-framework/src/main/scala/services/cdm/CdmTable.scala:41]: How about PartialFunc here to deconstruct the `arg` parameter?

**Review by s-vitaliy — COMMENTED** [review]
Added a comment related to readability

File: framework/arcane-framework/src/test/scala/models/CdmTableTests.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import com.azure.storage.common.StorageSharedKeyCredential
+import services.cdm.{CdmTable, CdmTableSettings}
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.must.Matchers
+import org.scalatest.matchers.should.Matchers.should
+
+import java.time.{OffsetDateTime, ZoneOffset}
+
+
+class CdmTableTests extends AsyncFlatSpec with Matchers {
```
Comment (s-vitaliy) [inline framework/arcane-framework/src/test/scala/models/CdmTableTests.scala:16]: Did I get it correct that this test should be executed manually against an existing storage container? If yes, would be nice to mention it in the class javadoc

**Review by s-vitaliy — COMMENTED** [review]
LGTM in general added one question.

File: framework/arcane-framework/src/test/scala/models/CdmTableTests.scala (hunk 2 above)
Comment (george-zubrienko) [reply framework/arcane-framework/src/test/scala/models/CdmTableTests.scala:16]: only the first two tests. I'll add tests that do not rely on Azure but rather test read model + read csv functionality instead

**Pull request merged**